workers/unity/Packages/io.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -119,27 +151,37 @@ private void ProcessNewEntities()
                 (Entity entity, ref SpatialEntityId spatialEntityId, ref Metadata.Component metadata) =>
                 {
                     var entityType = metadata.EntityType;
-                    var expectedTypes = entityTypeExpectations.GetExpectedTypes(entityType);
+                    var registered = entityTypeExpectations.TryGetValue(entityType, out var tuple);
+                    var (representation, expectedTypes) = tuple;
 
-                    foreach (var expectedType in expectedTypes)
+                    if (registered)
```

Comment:
I feel like there must be a way to refactor this lambda to make it clearer.

The if-endif in the middle really throws off the flow. I had to _really_ dissect this to figure out why the logic is laid out the way it is. 

Moving the if-endif to a function with a `[Conditional]` attribute would allow us to organise this into:

```
if (!registered)
{
	SetName();
	PostUpdateCommands.AddComponent(entity, new GameObjectInitSystemStateComponent { ... });
	return;
}

foreach (var expectedType in expectedTypes)
{
	// Do the checks.
}

SetName();

try
{
	// Resolve and call usercallback
}
catch (Exception e)
{
	Debug.LogException(e);
}

PostUpdateCommands.AddComponent(entity, new GameObjectInitSystemStateComponent { ... });
```

Which I feel captures the intent of this more clearly?